ibis/bigquery/client.py

```diff
@@ -90,11 +87,8 @@ def bigquery_field_to_ibis_dtype(field):
 @sch.infer.register(bq.table.Table)
 def bigquery_schema(table):
     pairs = [(el.name, dt.dtype(el)) for el in table.schema]
-    try:
-        if table.list_partitions():
-            pairs.append((NATIVE_PARTITION_COL, dt.timestamp))
-    except BadRequest:
-        pass
+    if table.partitioning_type is not None:
+        pairs.append((NATIVE_PARTITION_COL, dt.timestamp))
```

Comment:
Note: this won't be correct for tables that use the new [column-based time partitioning](https://cloud.google.com/bigquery/docs/creating-column-partitions).

Unfortunately google-cloud-bigquery still doesn't have the partitioning field property yet https://github.com/GoogleCloudPlatform/google-cloud-python/issues/4658 Whenever we add that there, this logic should change to only add the NATIVE_PARTITION_COL when the partitioning field is None.

Reply:
Is there a surefire way (even if either is a workaround) to 1) tell if a table is partitioned, and 2) what column(s) are being used for partitioning?

Reply:
Update: I see that there's some more info in the issue, I'll read through that :)